Fixes #14806 - added executors_count option

ShimShtein opened the pull request:
Now you can start multiple executors using `dynflow-executor`
or `foreman-task` and setting environment variable
`EXECUTORS_COUNT` to the desired number of executors.

adamruzicka commented on lib/foreman_tasks/dynflow/daemon.rb:
```diff
@@ -41,20 +42,22 @@ def run_background(command = "start", options = {})
 
-      Daemons.run_proc(options[:process_name],
+      options[:executors_count].times do
+        Daemons.run_proc(options[:process_name],
```
Each daemon needs to have unique process name. I'm getting the following error
```
bundle exec ../foreman-tasks/bin/dynflow-executor -c 2 start
Dynflow Executor: start in progress
dynflow_executor: process with pid 348 started.
ERROR: there is already one or more instance(s) of the program running
```

Editing the code to add the current iteration to the process name I got it to spawn several executors. The downside is the user has to remember how many executor he has if he wants to stop them. I'd expect stop would stop all executors
```
# bundle exec ../foreman-tasks/bin/dynflow-executor -c 2 start
Dynflow Executor: start in progress
dynflow_executor-0: process with pid 7266 started.
dynflow_executor-1: process with pid 7278 started.

# ps -e -o pid,cmd | grep executor
 7264 dynflow_executor-0_monitor
 7266 dynflow_executor-0
 7278 dynflow_executor-1
 7282 dynflow_executor-1_monitor

# bundle exec ../foreman-tasks/bin/dynflow-executor stop                                                                                                                                   
Dynflow Executor: stop in progress
dynflow_executor-0: trying to stop process with pid 7266...
dynflow_executor-0: process with pid 7266 successfully stopped.

# ps -e -o pid,cmd | grep executor                      
 7278 dynflow_executor-1
 7282 dynflow_executor-1_monitor
```

ShimShtein commented:
> Each daemon needs to have unique process name.

Fixed the multiple daemons problem - forgot to add a line that allows multiple instances. Now the naming is managed by the daemons gem.

>The downside is the user has to remember how many executor he has if he wants to stop them. I

As long as you don't change the number of executors, the stop command will stop the same amount of workers.

> the issue mentioned having the executor count configurable by the installer. Have you thought how this could be done?

Sure - I have introduced environment variable `EXECUTORS_COUNT` you can set it in the installer.
Example:
```
EXECUTORS_COUNT=10 bundle exec ../foreman-tasks/bin/dynflow-executor stop
```

adamruzicka commented:
Looks good to me

---
btw
> As long as you don't change the number of executors, the stop command will stop the same amount of workers.

With `:multiple => true` it ignores the provided count when stopping and always stops all the processes.

iNecas commented:
Works as expected. Thanks @ShimShtein
